strawberry/types/fields/resolver.py

```diff
@@ -0,0 +1,71 @@
+from typing import Callable, Optional, Set, TypeVar
+
+from cached_property import cached_property
+
+from strawberry.types import StrawberryArgument, StrawberryObject, \
+    StrawberryType
+
+T = TypeVar("T")
+
+
+class StrawberryResolver(StrawberryType[T]):
+    def __init__(self, func: Callable[..., T], *, description: Optional[str]):
+        self.wrapped_func = func
+        self._description = description
+
+    def __call__(self, *args, **kwargs) -> T:
+        # TODO: Should *args/**kwargs be some special object, or raw values?
+
+        if self._is_func_bound:
+            ...
+        if self._has_root_arg:
+            ...
+        if self._has_info_arg:
+            ...
+
+        # Deal with default args
+
+        return self.wrapped_func(*args, **kwargs)
+
+    @cached_property
+    def arguments(self) -> Set[StrawberryArgument]:
+        # TODO: Implement
+        ...
+
+    @property
+    def description(self) -> Optional[str]:
+        # TODO: Do resolvers get descriptions?
+        return self._description
+
+    @property
+    def name(self) -> str:
+        # TODO: What to do if resolver is a lambda?
+        return self.wrapped_func.__name__
```

Comment:
do we need the name here?

Reply:
I use the resolver's name as the wrapping field's name if the field didn't get an explicitly set name